lib/collection/url.js

```diff
@@ -400,6 +400,13 @@ _.assign(Url, /** @lends Url */ {
 
         // extract the hash
         p.hash = _.get(url.match(regexes.extractSearch), GET_1);
+
+        p.variable = _.transform(p.path, function (res, segment) {
+            if (_.startsWith(segment, PATH_VARIABLE_IDENTIFIER) && segment.length > 1) {
```

Comment:
@codenirvana - i know you have explained this to me once... but I keep forgetting the nature of the change added by `segment.length > 1` -> can you document it inline so that in future it is more clear why we did this.